src/api.cpp

```diff
@@ -4809,29 +4841,37 @@ cl_int CLVK_API_CALL clGetSupportedImageFormats(cl_context context,
             "  buffer : %s",
             vulkan_format_features_string(properties.bufferFeatures).c_str());
 
-        cvk_debug("  required format features %s",
+        cvk_debug("Required format features %s",
                   vulkan_format_features_string(required_format_feature_flags)
                       .c_str());
         VkFormatFeatureFlags features;
-        // TODO support linear tiling
-        features = properties.optimalTilingFeatures;
-
-        auto channel_order_str =
-            cl_channel_order_to_string(clfmt.image_channel_order);
-        auto channel_type_str =
-            cl_channel_type_to_string(clfmt.image_channel_data_type);
+        if (image_type == CL_MEM_OBJECT_IMAGE1D_BUFFER) {
+            features = properties.bufferFeatures;
+        } else {
+            // TODO support linear tiling
+            features = properties.optimalTilingFeatures;
+        }
         if ((features & required_format_feature_flags) ==
             required_format_feature_flags) {
+            VkComponentMapping components_sampled, components_storage;
+            get_component_mappings_for_channel_order(clfmt.image_channel_order,
+                                                     &components_sampled,
+                                                     &components_storage);
+            if ((components_sampled != components_storage) &&
+                (image_type == CL_MEM_OBJECT_IMAGE1D_BUFFER)) {
+                continue;
+            }
```

Comment:
I'm a bit confused by this condition. Those swizzles are unused for 1D buffer images. Why is this needed?

Reply:
I am not sure, this is coming from your initial commit I believe. Thinking about it, I don't see a reason to have it, maybe we should get rid of it?

Reply:
yeah, if we pass the CTS without it, then we should remove it. If we somehow need this to pass the CTS, we should find a clearer expression of the conditions.

Reply:
It seems that this test is needed. Otherwise the CTS thinks that we support `CL_INTENSITY` and `CL_LUMINANCE` with 1d buffer, which does not work with the actual implementation. As they are not mandatory, it feels ok to remove them with this test. What do you think?

Reply:
If they're not supported we should remove them but the condition used to do that is a hack (you said it was part of my draft patch, I'm assuming I'd done that as a quick way of excluding those formats). I've added an entry to #625 so we don't forget to revisit this.